### ISSUE TITLE
Webtransport client/server: (PANIC) `the message is not part of the protocol`

### ISSUE THREAD

**Issue by simbleau** [cBournhonesque/lightyear#483]
I have a minimum reproducible example, but I'm not sure at all what's going on here.

Debugging I've done:
- Yes, I have added the client/server plugins *before* the Protocol plugin
- Copied every setting from the shared plugins/example verbatim
- Made this minimum example
- Modified the lightyear code to determine the panic happens in `MessageRegistry::serialize`
- Pulled my hair out

<img width="1496" alt="image" src="https://github.com/cBournhonesque/lightyear/assets/48108917/e288a7f9-c910-4689-ba4e-d3babffbc12e">

The reproducible example is in my fork, branch [`minimum_example`](https://github.com/simbleau/lightyear/tree/minimum_example)

**Comment by simbleau** [comment]
UPDATE: This was a *nightmare* to find, but I found the root cause of the issues!!

Adding this to the minimum example,
```rust
        .add_systems(Startup, |mut commands: Commands| {
            commands.connect_client();
        })
```

It suddenly works.

Why would this be reason messages suddenly become "part of the protocol"?

Under the hood, `.connect_client()` does the following:
<img width="861" alt="image" src="https://github.com/cBournhonesque/lightyear/assets/48108917/e03d5a89-ed3c-48cd-b56c-dfde428e99c9">

This doesn't make sense to me, but hopefully we can fix this ASAP!

**Comment by simbleau** [comment]
What is the difference between `connection_manager.connect().unwrap();` and `commands.connect_client();`?

That should also be de-mystified, or have only 1 way to connect. This feels like a footgun having two ways to connect?

**Comment by cBournhonesque** [comment]
This should be fixed in the `cb/0.16` branch, now the `ConnectionManager` is rebuilt at the very end of starting app, after the protocol has been added.

Also added more comments so that users use `commands.connect_client()`.
I might end up making `connection_manager.connect().unwrap();` private, as you suggest
